### DISCUSSION TITLE
How to Run the Code

### DISCUSSION THREAD

**Discussion by martinbaker** [fdilke/bewl#8] [General]
I have tried running Bewl using various methods and so far I have not been successful. I don't really have any practical experience with scala, could you point me in the right direction?

First I tried running from the command line console as follows:

```
git clone https://github.com/fdilke/bewl.git
cd bewl
sbt console
```

The first two lines worked but the sbt command generated errors.

```
mjb@localhost:~/bewl> sbt console
[info] [launcher] getting org.scala-sbt sbt 0.13.18  (this may take some time)...
:: loading settings :: url = jar:file:/usr/share/java/ivy.jar!/org/apache/ivy/core/settings/ivysettings.xml
:: loading settings :: url = jar:file:/usr/share/java/ivy.jar!/org/apache/ivy/core/settings/ivysettings.xml
:: loading settings :: url = jar:file:/usr/share/java/ivy.jar!/org/apache/ivy/core/settings/ivysettings.xml
:: loading settings :: url = jar:file:/usr/share/java/ivy.jar!/org/apache/ivy/core/settings/ivysettings.xml
:: loading settings :: url = jar:file:/usr/share/java/ivy.jar!/org/apache/ivy/core/settings/ivysettings.xml
:: loading settings :: url = jar:file:/usr/share/java/ivy.jar!/org/apache/ivy/core/settings/ivysettings.xml
:: loading settings :: url = jar:file:/usr/share/java/ivy.jar!/org/apache/ivy/core/settings/ivysettings.xml
:: loading settings :: url = jar:file:/usr/share/java/ivy.jar!/org/apache/ivy/core/settings/ivysettings.xml

:: problems summary ::
:::: WARNINGS
                ::::::::::::::::::::::::::::::::::::::::::::::

                ::              FAILED DOWNLOADS            ::

                :: ^ see resolution messages for details  ^ ::

                ::::::::::::::::::::::::::::::::::::::::::::::

                :: com.jcraft#jsch;0.1.50!jsch.jar

                ::::::::::::::::::::::::::::::::::::::::::::::



:: USE VERBOSE OR DEBUG MESSAGE LEVEL FOR MORE DETAILS
[error] [launcher] download failed: com.jcraft#jsch;0.1.50!jsch.jar
[error] [launcher] error during sbt launcher: error retrieving required libraries
  (see /home/mjb/.sbt/boot/update.log for complete log)
[error] [launcher] could not retrieve sbt 0.13.18
```

Does sbt need to be setup in some way? the only other issue I can think of is that opensuse that I am running appears to come with java runtime environment (java-11-openjdk - OpenJDK 11) but not the full JDK from Oracle. Is this required? I don't know.

I then tried to run from Eclipse (which comes with a JDK). However I could not install Scala in Eclipse due some bug as explained here: https://www.eclipse.org/forums/index.php/t/1111325/

I then tried running scala in JetBrains as explained on this page: https://www.euclideanspace.com/maths/discrete/category/topoi/bewl/jetBrains/index.htm. This appeared to install all the files correctly but there were build errors in many of the scala files.

Martin

**Comment by fdilke** [comment]
Hi! Only just saw this, clearly I need better GitHub notifications :)

Delighted you are trying to run Bewl, I'll do what I can to help.

It should work with any reasonably up-to-date JDK. The Java version I'm using is:

OpenJDK 64-Bit Server VM (build 18.0.1.1+2-6, mixed mode, sharing)

But would expect it to work with anything from 11 on, or maybe as far back as 7.

Maybe the issue is your SBT version? I'm using 1.9.6, although sbt should automatically use the right version from build.properties.

I'd also be interested to know your use case. Are you just generally interested in  doing category theory in software, or do you have a specific topos calcuilation in mind?

The most generally applicable 'heavy lifting' Bewl is currently capable of would be in the topos of actions over a monoid, which might be of interest if you're studying semigroup/monoid theory. It can also be handy for group and permutation calculations, but those may be adequately covered for you by standard computer algebra packages.

Best
Felix

**Comment by fdilke** [comment]
Sorry, I have not been able to devote much time to this, but you might also want to try running Bewl 2 which is being more actively developed. It's in the **bewl2** subdirectory of the main Bewl repo.
